pyalertme/zbnode.py

```diff
@@ -475,6 +477,9 @@ def receive_message(self, message):
                 self.send_message(reply, source_addr_long, source_addr_short)
                 time.sleep(0.5)
 
+            # Update any attributes which may need updating
```

Comment:
Ref #55 Glad this seems to behave better. However with the moving of this it looks like we have broken the unittest test_mock_serial.  I think we have broken it for the better but please can you look at fixing this also?

I think because we have, in effect, altered the order in which replied are sent (knowingly) we will need to update the unit test to suit. The test which is failing is actually a test of the mock_serial class, but it used this function as part of its test.

If you goto 
https://travis-ci.org/jamesleesaunders/PyAlertMe/builds/294800979?utm_source=github_status&utm_medium=notification

You will see the failed test. Otherwise run 'nosetests' (or pytest) to see the errors yourself.

Reply:
Jim,
I will try and look at it this weekend.  I got slammed this week with other commitments.
Regards,
 James

Reply:
Hi James, thanks for fixing the test :-) all looks good. I’ll get this merged in.